src/gui/providers/qgspointcloudsourceselect.cpp

```diff
@@ -180,4 +270,194 @@ void QgsPointCloudSourceSelect::showHelp()
 }
 
+void QgsPointCloudSourceSelect::clearOpenOptions()
+{
+  mOpenOptionsWidgets.clear();
+  mOpenOptionsGroupBox->setVisible( false );
+  mOpenOptionsLabel->clear();
+  while ( mOpenOptionsLayout->count() )
+  {
+    QLayoutItem *item = mOpenOptionsLayout->takeAt( 0 );
+    delete item->widget();
+    delete item;
+  }
+}
+
+void QgsPointCloudSourceSelect::fillOpenOptions()
+{
+  clearOpenOptions();
+  if ( mDataSourceType.isEmpty() )
+  {
+    return;
+  }
+
+  const QUrl url = QUrl::fromUserInput( mPath );
+  const std::string driverName = pdal::StageFactory::inferReaderDriver( url.fileName().toStdString() );
+  // Only handle the readers.text at the moment.
+  // This could be expanded for other drivers.
+  if ( driverName != std::string( "readers.text" ) )
+  {
+    return;
+  }
+
+  pdal::StageFactory stageFactory( false );
+  pdal::Stage *stage = stageFactory.createStage( driverName );
+  if ( !stage )
+  {
+    QgsDebugMsgLevel( QStringLiteral( "Unable to create stage for driver %1" ).arg( QString::fromStdString( driverName ) ), 2 );
+    return;
+  }
+
+  pdal::ProgramArgs args;
+  stage->addAllArgs( args );
+  std::stringstream args_json_stream;
+  args.dump3( args_json_stream );
+
+  json args_json;
+  args_json_stream >> args_json;
+
+  const QList<QString> ignoredArgs = {
+    QStringLiteral( "count" ),
+    QStringLiteral( "default_srs" ),
+    QStringLiteral( "filename" ),
+    QStringLiteral( "log" ),
+    QStringLiteral( "option_file" ),
+    QStringLiteral( "user_data" ),
+  };
+
+  for ( auto &elem : args_json.items() )
+  {
+    auto arg = elem.value();
+    const std::string argName = arg["name"];
+    if ( std::find( ignoredArgs.begin(), ignoredArgs.end(), QString::fromStdString( argName ) ) != ignoredArgs.end() )
+    {
+      continue;
+    }
+
+    QLabel *optionLabel = new QLabel( argName.c_str() );
+    QWidget *optionControl;
+
+    std::string argDefault = arg.value( "default", "" );
+    // skip default value may be empty. Use "0" instead to create a spinbox widget
+    if ( argDefault.empty() && argName == std::string( "skip" ) )
+    {
+      argDefault = "0";
+    }
+    bool parsedInt;
+    const int defaultIntValue = QString::fromStdString( argDefault ).toInt( &parsedInt );
+
+    if ( argName == "separator" )
+    {
+      QWidget *separatorContainer = createSeparatorOptionWidget();
+      if ( parsedInt )
+      {
+        QComboBox *comboBox = separatorContainer->findChild<QComboBox *>();
+        if ( comboBox )
+        {
+          const int index = comboBox->findData( QVariant( defaultIntValue ) );
+          if ( index != -1 )
+          {
+            comboBox->setCurrentIndex( index );
+          }
+        }
+      }
+      optionControl = separatorContainer;
+    }
+    else if ( argName == std::string( "override_srs" ) )
+    {
+      optionControl = new QgsProjectionSelectionWidget();
+    }
```

Comment:
I think we should NOT be exposing this option. QGIS already has settings to override a layer's CRS in a standard way which works for all different layer types, and users should be using that instead.